src/figuresClasses.ts

```diff
@@ -1,11 +1,72 @@
-export interface Figure {}
+export interface Figure {
+  shape: 'triangle' | 'rectangle' | 'circle';
+  color: 'red' | 'green' | 'blue';
+  getArea: () => number;
+}
+
+export class Triangle implements Figure {
+  shape: Figure['shape'] = 'triangle';
+
+  constructor(
+    public color: Figure['color'],
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (
+      this.a + this.b <= this.c ||
+      this.a + this.c <= this.b ||
+      this.b + this.c <= this.a
+    ) {
+      throw new Error('not triangle');
+    } else if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('not triangle');
```

Comment:
The condition `a <= 0 || b <= 0 || c <= 0` should use `this.a`, `this.b`, and `this.c` instead of `a`, `b`, and `c` to correctly reference the instance variables.